src/java.base/share/classes/javax/crypto/Cipher.java

```diff
@@ -1048,13 +1048,13 @@ public final int getOutputSize(int inputLen) {
     /**
      * Returns the parameters used with this cipher.
      *
-     * <p>The returned parameters may be the same that were used to initialize
-     * this cipher, or may contain a combination of default and random
-     * parameter values used by the underlying cipher implementation if this
-     * cipher requires algorithm parameters but was not initialized with any.
+     * <p>If this cipher has been previously initialized with parameters,
+     * this method returns the same parameters. Otherwise, this method may
+     * return a combination of user-supplied, default and randomly generated
```

Comment:
> ... a combination of user-supplied, default and randomly generated parameter values ...

Other than init(), which APIs could impact the user-supplied parameters?  Is it the getInstance()?

Is it OK to just use the provider-specific default values instead, by replacing the "a combination of user-supplied, default and randomly generated ..."?

Reply:
I am thinking just init() as it takes parameters.
Let me take a second look at various Cipher algorithm implementations and see if this can be further pared down.

Reply:
Please see the updated wording in earlier comments.